DYDX-Bots/Perpetual-Large-Deposits-Withdrawals/README.md

```diff
@@ -0,0 +1,85 @@
+# Large deposits withdrawals on perpetual exchange
+
+## Description
+
+This bot detects deposits and withdrawals on dYdX perpetual exchange contract with large transfered amounts.
+
+> The bot can operate in two different modes, used to determine the threshold of a _large_ transfered amount.
+>
+> - `STATIC` mode refers to the bot using a static predefined threshold.
+> - `PERCENTAGE` refers to setting the threshold as a percentage of the contract balance.
+
+> In order to switch between the two modes, change `DYNAMIC_CONFIG` to `STATIC_CONFG` in agent.ts.
+
+## Supported Chains
+
+- Ethereum
+
+## Alerts
+
+- DYDX-1-1
+
+- Fired when `LogDeposit` event is emitted on dYdX perpetual contract with a large quantizied amount.
+- Severity is always set to "Info".
+- Type is always set to "Info".
+- Metadata contains:
+
+  - `quantizedAmount`: tokens amount that was deposited.
+  - `starkKey`: stark key of the user making the deposit.
+  - `token`: the token that was transfered.
+
+- DYDX-1-2
+
+  - Fired when `LogWithdrawalPerformed` event is emitted on dYdX perpetual contract with a large quantizied amount.
+  - Severity is always set to "Info"
+  - Type is always set to "Info"
+  - Metadata contains:
+    - `quantizedAmount`: tokens amount that was withdrawn.
+    - `token`: the token that was transfered.
+    - `recipient`: ethereum address receiving the tokens.
+    - `ownerKey`: stark key of user withdrawing his tokens.
+
+- DYDX-1-3
+
+  - Fired when `LogMintWithdrawalPerformed` event is emitted on dYdX perpetual contract with a large quantizied amount.
+  - Severity is always set to "Info".
+  - Type is always set to "Info".
+  - Metadata contains:
+    - `quantizedAmount`: tokens amount that was withdrawn.
+    - `token`: the asset that was transfered.
+    - `assetId`: Id of the minted asset.
+    - `ownerKey`: stark key of the recipient.
+
+- DYDX-1-4
+
+  - Fired when one of the events is emitted with an asset different from the system asset.
+  - Severity is always set to "High".
+  - Type is always set to "Suspicious".
+  - Metadata contains:
+    - `quantizedAmount`: tokens amount that was deposited or withdrawn.
+    - `starkKey`: stark key of the user involved in the tranfer.
+    - `assetType`: assetType that is different from the system one.
+
+## Test Data
+
+The bot behaviour can be verified with the following transactions, using the default configurations:
+
+### Mainnet
+
+The following transactions return a finding with the `STATIC` mode only.
+
+- 0x30c630fe50e492eeff514cab7590921ab21c4fc224cfff7bad0f82ee2a47db0f (`LogWithdrawalPerformed`)
+- 0x77387abbc97753343fda4008b543ef01d8ccd987b887d20c047b748e3894e8cb ( `LogDeposit`)
+
+### Kovan Testnet
```

Comment:
Maybe adding a note that these same transactions will work with either "mode?" `STATIC` or `PERCENTAGE`.